src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingStackTraceService.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.API
+{
+    internal sealed class UnitTestingStackTraceService
+    {
+        private readonly Workspace _workspace;
+        private readonly IStackTraceExplorerService _stackTraceService;
```

Comment:
📝 The service accessor should only have an instance of the service unless there is no way to implement it otherwise.

```suggestion
        private readonly IStackTraceExplorerService _stackTraceService;
```